### DISCUSSION TITLE
Is there an analogue of `mapped_column(sort_order=...)` that I can use inside `@declared_attr`?

### DISCUSSION THREAD

**Discussion by slava-re** [sqlalchemy/sqlalchemy#11249] [Usage Questions] [answered]
Good day to you!

Running into a question where we have some columns in our ORM model super-classes and mixins defined using `@declared_attr`. E.g.

```python
    @declared_attr
    def tenant_oid(cls):
        return Column(
            ForeignKey("tenant.oid", name=f"{cls.__tablename__}_tenant_oid_fkey"), # Want to use `cls` here
            ...
        )
```

And there seemingly being no ability to control the order of such columns. I understand how such a feature could be tricky to implement depending on how `declared_attr` is done, so this is not a strong feature request, just wanted to ask if maybe there's an existing API I am missing which would interact with other `mapped_column(sort_order=...)` arguments so that I could order the columns properly amongst themselves.

**Comment by zzzeek** [comment, marked as answer]
you can (and maybe even should) return `mapped_column()` from your `@declared_attr` , does that help?

**Reply by slava-re** [reply]
Yep, this works as intended. All the docs on `@declared_attr` show returning `Column`, so I assumed returning `mapped_column` is not supported. Thank you!!

**Reply by CaselIT** [reply]
The docs probably need to be updated. If you could send a PR it would be greatly appreciated!
